### DISCUSSION TITLE
SSO and OAuth greatly needed and it's safe!

### DISCUSSION THREAD

**Discussion by vlev1n** [flet-dev/flet#1385] [Ideas]
In Feb 2022 OAuth support is removed from Flet and the author thought it does not make sense #985.

IMHO OAuth does make sense cuz there are many ways developers can make auth work WITHOUT Client ID and Client Secret! Many Chinese apps like WeChat, QQ and Feishu use this way. The secret is safely stored in an "intermediary server", which is between the client and the third party platform and the client only need to send its Client ID. [Feishu's approach](https://open.feishu.cn/document/common-capabilities/sso/api/get-access_token?lang=en-US)

I tried to submit a PR but found the code complicated and I'm afraid of the side-effect. I hope in the future version the author could bring back SSO and OAuth and this deed will obviously push Flet to wider usage scenario and be more developers' choice.

By the way developers has started to fork and bring back OAuth and SSO by themselves. See [this fork](https://github.com/hokaso/fork-flet)

**Comment by hokaso** [comment]
thanks bro~ Especially in China, we use various sso platforms very widely, such as WeChat, qq, etc., and there are multiple platforms under my company, all of which require dependency sso login, including the client-end developed based on the flet framework recently, which I also want to include.

**Comment by FeodorFitsner** [comment]
SSO is not removed or deprecated, it's just disabled in a single-use Desktop mode when Python program is run with built-in socket server.

To get SSO working with desktop client you have to force Flet HTTP web server, with:

```python
ft.app(main, port=8550, view=ft.WEB_BROWSER)
```

and then call the app as:

```
flet run <your-program>.py
```

**Reply by hokaso** [reply]
oh, thank you! It helps a lot~

**Reply by hokaso** [reply]
> SSO is not removed or deprecated, it's just disabled in a single-use Desktop mode when Python program is run with built-in socket server.
> 
> To get SSO working with desktop client you have to force Flet HTTP web server, with:
> 
> ```python
> ft.app(main, port=8550, view=ft.WEB_BROWSER)
> ```
> 
> and then call the app as:
> 
> ```
> flet run <your-program>.py
> ```

oh, thank you! It helps a lot~

**Reply by hokaso** [reply]
> SSO is not removed or deprecated, it's just disabled in a single-use Desktop mode when Python program is run with built-in socket server.
> 
> To get SSO working with desktop client you have to force Flet HTTP web server, with:
> 
> ```python
> ft.app(main, port=8550, view=ft.WEB_BROWSER)
> ```
> 
> and then call the app as:
> 
> ```
> flet run <your-program>.py
> ```
I apologize, I have diligently implemented your suggestion. However, I have noticed that the application is presented as a web UI rather than a window software in the form of an exe as you described. Is there any possibility to package it as an exe file and run it as a window software while ensuring compatibility with SSO login?

**Comment by FeodorFitsner** [comment]
OK, so we need a way to force using web server, but still run it with desktop UI!

**Reply by FeodorFitsner** [reply]
https://github.com/flet-dev/flet/issues/1401
